File: server/dna.core/dna.core.service/Models/FirebaseUserMapModel.cs
using dna.core.data.Infrastructure;
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models
{
    public class FirebaseUserMapModel : IModelBase
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string DeviceId { get; set; }
        public string DeviceToken { get; set; }
        public data.Infrastructure.OperatingSystem OperatingSystem { get; set; }

        public virtual UserModel User { get; set; }
    }
}

File: server/medicore.apps/src/model/FirebaseUserMapModel.ts
import { Type } from 'class-transformer';
import { IEntityBase } from './IEntityBase';
import { OperatingSystem, UserModel} from '../model';

export class FirebaseUserMapModel implements IEntityBase {
    id: number;
    userId: number;
    deviceId: string;
    deviceToken: string;
    operatingSystem: OperatingSystem;

    //user model
    @Type(() => UserModel)
    user: UserModel;

    constructor() {
        this.id = 0;
        this.userId = 0;
        this.deviceId = null;
        this.deviceToken = null;
        this.operatingSystem = null;
        this.user = null;
    }
}